pkgs/development/libraries/bashup-events/default.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv, lib, callPackage, runCommand, resholvePackage, fetchFromGitHub, bash, shellcheck, doCheck ? true, doInstallCheck ? true }:
+let
+  # extracting this so that it's trivial to test in other shells
+  installCheck = shell:
+    ''
+      echo "testing bashup.events in ${shell}"
+      ${shell} <<'EOF'
+      source $out/bin/bashup.events
+      neat(){
+        echo $0: Hi from event \'test event\'. I can have both $1 and $2 arguments.
+        exit 0
+      }
+      event on "test event" @2 neat curried
+      echo event registered
+      event emit "test event" runtime
+      exit 1 # fail if emitting event didn't exit clean
+      EOF
+    '';
+  common = { variant, version, branch, rev, hash, fake ? false, keep }: resholvePackage rec {
```

Comment:
I would prefer if you moved this into a generic.nix and create a default.nix for 4.4 and a 3.2.nix. This would keep the size of this file down a bit and should make things easier in the future.

Reply:
done in e586d3ff8bda9a21a9c05f4db30463f6240f4da9